cvss40.js

```diff
@@ -0,0 +1,1031 @@
+      "IR":  ["X", "H", "M", "L"],
+      "AR":  ["X", "H", "M", "L"],
+      "MAV": ["X", "N", "A", "L", "P"],
+      "MAC": ["X", "L", "H"],
+      "MAT": ["X", "N", "P"],
+      "MPR": ["X", "N", "L", "H"],
+      "MUI": ["X", "N", "P", "A"],
+      "MVC": ["X", "H", "L", "N"],
+      "MVI": ["X", "H", "L", "N"],
+      "MVA": ["X", "H", "L", "N"],
+      "MSC": ["X", "H", "L", "N"],
+      "MSI": ["X", "S", "H", "L", "N"],
+      "MSA": ["X", "S", "H", "L", "N"],
+      // Supplemental (6 metrics)
+      "S":  ["X", "N", "P"],
+      "AU": ["X", "N", "Y"],
+      "R":  ["X", "A", "U", "I"],
+      "V":  ["X", "D", "C"],
+      "RE": ["X", "L", "M", "H"],
+      "U":  ["X", "Clear", "Green", "Amber", "Red"],
+  };
+
+  static MAX_COMPOSED = {
+      // EQ1
+      "eq1": {
+          0: ["AV:N/PR:N/UI:N/"],
+          1: ["AV:A/PR:N/UI:N/", "AV:N/PR:L/UI:N/", "AV:N/PR:N/UI:P/"],
+          2: ["AV:P/PR:N/UI:N/", "AV:A/PR:L/UI:P/"]
+      },
+      // EQ2
+      "eq2": {
+          0: ["AC:L/AT:N/"],
+          1: ["AC:H/AT:N/", "AC:L/AT:P/"]
+      },
+      // EQ3+EQ6
+      "eq3": {
+          0: { "0": ["VC:H/VI:H/VA:H/CR:H/IR:H/AR:H/"], "1": ["VC:H/VI:H/VA:L/CR:M/IR:M/AR:H/", "VC:H/VI:H/VA:H/CR:M/IR:M/AR:M/"] },
+          1: { "0": ["VC:L/VI:H/VA:H/CR:H/IR:H/AR:H/", "VC:H/VI:L/VA:H/CR:H/IR:H/AR:H/"], "1": ["VC:L/VI:H/VA:L/CR:H/IR:M/AR:H/", "VC:L/VI:H/VA:H/CR:H/IR:M/AR:M/", "VC:H/VI:L/VA:H/CR:M/IR:H/AR:M/", "VC:H/VI:L/VA:L/CR:M/IR:H/AR:H/", "VC:L/VI:L/VA:H/CR:H/IR:H/AR:M/"] },
+          2: { "1": ["VC:L/VI:L/VA:L/CR:H/IR:H/AR:H/"] },
+      },
+      // EQ4
+      "eq4": {
+          0: ["SC:H/SI:S/SA:S/"],
+          1: ["SC:H/SI:H/SA:H/"],
+          2: ["SC:L/SI:L/SA:L/"]
+
+      },
+      // EQ5
+      "eq5": {
+          0: ["E:A/"],
+          1: ["E:P/"],
+          2: ["E:U/"],
+      },
+  };
+
+  // max severity distances in EQs MacroVectors (+1)
+  static MAX_SEVERITY = {
+      "eq1": {
+          0: 1,
+          1: 4,
+          2: 5
+      },
+      "eq2": {
+          0: 1,
+          1: 2
+      },
+      "eq3eq6": {
+          0: { 0: 7, 1: 6 },
+          1: { 0: 8, 1: 8 },
+          2: { 1: 10 }
+      },
+      "eq4": {
+          0: 6,
+          1: 5,
+          2: 4
+      },
+      "eq5": {
+          0: 1,
+          1: 1,
+          2: 1
+      },
+  };
+
+  static MACRO_VECTOR_DETAILS = {
+      "Exploitability": 0,
+      "Complexity": 1,
+      "Vulnerable system": 2,
+      "Subsequent system": 3,
+      "Exploitation": 4,
+      "Security requirements": 5
+  };
+
+  static MACRO_VECTOR_VALUES = {
+      "0": "High",
+      "1": "Medium",
+      "2": "Low",
+      "3": "None"
+  };
+
+
+  /**
+   * Constructs a CVSS40 object and initializes its properties.
+   *
+   * This constructor validates the provided CVSS v4.0 vector string against the CVSS v4.0 schema,
+   * extracts the metrics from the vector string, computes the equivalency vector,
+   * and calculates the base score.
```

Comment:
"base" is false, there is no "base" score notion in the CVSS v4.0 specification. The nomenclature helps understanding what metric groups were involved in computation.